Runtime/Scripts/RGBotConfigs/RGAction_KeyPress.cs

```diff
@@ -0,0 +1,298 @@
+                        {
+                            _inputActions.Add(iac.keyCode, inputActionControl);    
+                        }
+                        else if (inputActionControl is AnyKeyControl)
+                        {
+                            _anyKey = inputActionControl;
+                        }
+                    }
+                }
+            }
+#endif
+        }
+
+        public void Update()
+        {
+#if ENABLE_INPUT_SYSTEM
+            // one button click per frame update
+            if (_keysToPress.TryDequeue(out RG_KeyPress_Data keyToPress))
+            {
+                List<InputControl> actionsToTake = new();
+                // check for any key 
+                if (_anyKey != null)
+                {
+                    actionsToTake.Add(_anyKey);
+                }
+
+                var theKey = _inputActions.GetValueOrDefault(keyToPress.keyId, null);
+                if (theKey != null)
+                {
+                    actionsToTake.Add(theKey);
+                }
+
+                foreach (var control in actionsToTake)
+                {
+                    PressKey(keyToPress.keyId, control, keyToPress.holdTime);
+                }
+            }
+            
+            // handle un-pressing any keys that have expired their time
+            var keys = keysDown.Keys.ToList();
+            foreach (var key in keys)
+            {
+                // this is < instead of <= so that keys last at least 1 frame
+                if (keysDown[key] < Time.unscaledTime)
+                {
+                    // dictionary will have this value because we clicked it
+                    UnPressKey(key, _inputActions.GetValueOrDefault(key));
+                }
+            }
+
+            if (keysDown.Count == 0 && _anyKey != null)
+            {
+                // un-press the 'any' key
+                UnPressKey(null, _anyKey);
+            }
+            
+            int id = gameObject.transform.GetInstanceID();
+            // render debug text
+            if (renderDebugGizmos)
+            {
+                var debugText = "";
+                var keysList = keysDown.ToList();
+                keysList.Sort((a,b) => a.Key-b.Key);
+                foreach (var (key,value) in keysList)
+                {
+                    debugText += $"{key.ToString()} : {(value-Time.unscaledTime):F2}\r\n";
+                }
+
+                
+                if (string.IsNullOrEmpty(debugText))
+                {
+                    RgGizmos.DestroyText(id);
+                }
+                else
+                {
+                    RgGizmos.CreateText(id, debugText, debugGizmosOffset);
+                }
+
+                // force updating these now
+                RgGizmos.OnDrawGizmos();
+            }
+            else
+            {
+                RgGizmos.DestroyText(id);
+                // force updating these now
+                RgGizmos.OnDrawGizmos();
```

Comment:
I could move this call into OnDrawGizmos in this same class I suppose.  That would be more correct

Reply:
Yeah you'd basically just want to move the RgGizmos.OnDrawGizmos(); call to the monob's OnDrawGizmos functon, but keep everything else in Update since OnDrawGizmos is only called when the Gizmo feature is turned on.